### ISSUE TITLE
Some To-dos!

### ISSUE THREAD

**Issue by LNJplus** [BlockMen/minetest_next#33]
I asked some friends (okay two) what is missing in Minetest (NeXt), that are the results:
* 1. Carpets (wool) 
* 2. Improved Signs
* 3. Achivements
* 4. Skins (maybe modified version of player_textures, skins should be in textures/skins/character_<name>.png, if possible)
* 5. Furniture (a chair with sitting function and a table)
* 6. Some more player animations like sneaking and swimming (but I think that's not so easy or needs engine change)
* 7. Carts, maybe
* 8. Vines (simple but with growing and maybe a length limit)
* 9. Fence Gate (and maybe some more fancy fences)
* 10. "acacia_wood.png" shouldn't be so extreme red and maybe use the colourize function (for pine and junglewood too)
* 11. Randomly big and huge trees (idea by BlockMen)
* 12. Improve Creative Inventory (add place for armor, add search)
* 13. Add some nice backgrounds for the menus of: Inventory, Chests, Signs, ... (not only grey / black)
* 14. A Crafting Book where you can see all crafting recipe (you should get this at first join)
* 15. Maybe sounds and increased walk speed (just a little bit) when the player is exhausted (e.g. after long time underwater)

Just some ideas! :) What do you think about them? If you've got no idea what to do here is something to do!

:smile_cat:

**Comment by BlockMen** [comment]
I think the main focus should be on the improved creative inventory (12.) since adding more nodes (like the chairs, carpets, etc) would bloat it even more. Also Mobs (18.) should go in soon.

Things that should be done by engine changes are signs (2.) improved fences and fence gates (9.).

Currently there is a pull request to add hunger to the engine, so maybe this point is obsolete near future.

After quick reading of the TODO list i agree with most of it. Though each point has to be thought about in terms of performance, especially the mapgen related things (e.g. huge trees) that might have a big impact on generation time.

**Comment by kilbith** [comment]
1. Carpets : No, too specific and bloat the inventory and can't place anything over it.
2. Improved Signs : Yes, if only the texture is changed to something more neutral (don't add a damn entity for text).
3. Achievements : Why not.
4. Skins : No, this should be a client-side thing so that your skin is applied on every server.
5. Sit-able chair and table : Yes, already did in my mod (https://github.com/kilbith/xdecor/blob/master/sitting.lua)
6. More player animatios : Why not.
7. Carts : Yes, but only when the engine deals better with the entities.
8. Vines : Yes, but not growing - they shall be included in the jungletree schematic.
9. Fence Gate : Yes, also paired with Cobble Wall.
10. Acacia texture not too red : Yes.
11. Big trees : it may cause problems with lighting updates in the engine (see MoreTrees).
12. Improved Inventory : Done I think ?
13. Nicer formspec backgrounds : Yes, but not too high-resolution.
14. Craft guide book : Yes, there's a simple node-based crafting guide in my mod (https://github.com/kilbith/xdecor/blob/master/worktable.lua)
15. Increased walk speed : No, it's fine as-it. The engine should have a special running animation with an energy limit (MC-like).

**Comment by BlockMen** [comment]
@kilbith why are carpets too specific for you while chair and table are not?

**Comment by kilbith** [comment]
@BlockMen You're right to point a contradiction, but chair and table are basic furnitures, as well as universal / intemporal.

**Comment by BlockMen** [comment]
I don't see an objective difference between carpets and funiture. As long it is not bloating an inventory (e.g. by adding categories, here: wool) im fine with it.

**Comment by lnjX** [comment]
- 19. Use the hud textures from Wasteland

![screenshot_20160126_180135](https://cloud.githubusercontent.com/assets/13557034/12588001/f8e648da-c456-11e5-9452-ccf57f6e2b21.png)

(@BlockMen)  Would you agree to that? Then I can make a pull request.

**Comment by BlockMen** [comment]
I'm not sure about the hotbar, the icons can go in.

But notice that you need the hud mod aswell then, else you won't have the backgrounds/outlines.
